Add Malyan M300 / Monoprice Delta Mini support

mojocorp opened the pull request:
### Requirements

* Filling out this template is required. Pull Requests without a clear description may be closed at the maintainers' discretion.

### Description

Add support for the Malyan M300 / Monoprice Delta mini based on the STM32F070.
ststm32 6.1.0 minimum is required in order to get the M300 variant.

The last commit activate the autobuild but this should not be merged since the M300 has
only 123K flash and if the default config somehow activate too many features, it will exceed
this limit and mark the build has failed.

### Benefits

<!-- What does this fix or improve? -->

### Related Issues

<!-- Whether this fixes a bug or fulfills a feature request, please list any related Issues here. -->

mojocorp committed:
Add Malyan M300 / Monoprice Delta Mini support

mojocorp committed:
Add M300 test, auto-build support - DO NOT MERGE

The malyan M300 has only 123K flash and if the Marlin binary continue
to grow, the default config may already exceed this limit and make the
build fails.

mojocorp commented:
I just updated the PR since the required ststm32 6.1.0 was jsut released.

mojocorp commented:
@thinkyhead I just rename the pin E0_AUTO_FAN_PIN into AUTO_FAN_PIN and updated the conrresponding confirguration in order to make it more explicit and avoid potential mistakes.
I won't update this PR anymore, I promise :)

thinkyhead commented:
Since I can't edit this (you didn't give permission) I applied the differences here to #17421. You actually could have submitted a PR to the branch that #17421 is based on, and I think now GitHub even has an inline change-request thing for doing direct edits on files. Thoughts for future PRs!

mojocorp commented:
@thinkyhead the usual github workflow is to review and let the owner of the PR do the changes.
This way the owner really submit it's own work and is confident that it's working. When you amend a commit, you alter what the owner did but the name stay. It's exactly what happened with the configuration where my name appears on a commit I did not do and I did not agree with.
Now I can see that the M300 support was added under your name....It's fine, but that's a problem that does not appears in the usual github workflow.
Overall it took longer than necessary: my very first PR that was already working differs from 3 lines, and a single line was added by you. I could have addressed your remarks in a few minutes.

thinkyhead commented:
As mentioned, you can submit a PR to the branch of an existing PR, and when that PR is accepted it updates the corresponding PR. There's no need to amend. The only reason I generally amend or rebase a branch is that a PR is getting out of hand and it's hard to pick out changes from a long list of commits. And whenever I do so I paste the exact commands required to bring the working copy in sync. Marking a PR as not-editable only slows down the process and steals time from other issues.

mojocorp commented:
>As mentioned, you can submit a PR to the branch of an existing PR, and when that PR is >accepted it updates the corresponding PR.

There is no need for that if you just do a simple review and let the owner do the changes.

>There's no need to amend.

Yes!

>The only reason I generally amend or rebase a branch is that a PR is getting out of hand and it's >hard to pick out changes from a long list of commits. 

Again, why not just let the owner do the required changes ? I understand if you don't get any response, then just take over. In my case, you just forked my PR and closed it without saying anything.

>Normally, I simply add new commits, which is what most GitHub geeks are used to. 

That's perfectly fine to add commits in your name afterward.

>Marking a PR as not-editable only slows down the process and steals time from other issues.

Again, just let the PR owner address your review. It's easier to track the requested changes, the reason, people can learn in the process, etc...
I hope you will consider following the github workflow since it's a well working one!

thinkyhead commented:
> …just do a simple review and let the owner do the changes.
> Again, why not just let the owner do the required changes?

"Just." You must be new here. That simply doesn't work. Coders are sloppy and spell badly and leave stray spaces and tabs all over the place, and they tend to disappear.

Chasing down contributors for changes doesn't work with my schedule or workflow, and most of the time they don't follow directions. I have so many issues coming in that I must work on them unceasingly, and I make sure every PR is polished up before it is merged. When I do manage to find enough time and space to work on PRs and issues, I'm going to take that opportunity to finish up and merge them, because things must keep moving. I cannot do long email chains and ping people and nag them to fix their spelling and spacing.

That said, this conversation is exactly the kind of thing I don't care for.

> I hope you will consider following the github workflow since it's a well working one!

That comes across as condescending considering how much work I do every day and how exhausted I feel right now. What you are proposing — spending more time writing and discussing and nagging and less time working on code — would drive me up the wall.

mojocorp commented:
Oh I'm really honestly sorry that you feel it that way. It was really not intended to be condescending, sorry for the wrong formulation..
It was just a try to make the workflow more efficient and you are right, I don't have a large open source project, just coworkers that just do their job. I was just really surprised by the workflow and to be honest, your answer did not help me understand why its like that. But Im sure many open source contributors are willing to rework a PR. You are doing a very good job and very helpful for the community, i would never criticize that! Please accept my appologies and get a lot of courage to continue this amazing work !

thinkyhead commented:
I don't mean to be touchy, but the days are long and the issues are never-ending.

What I mean to say is that I do follow the GitHub workflow because it's the natural way, but I prefer it when contributors leave the box checked.

![image](https://user-images.githubusercontent.com/698003/79776461-63bdb280-82fb-11ea-8546-bff02665327c.png)

When contributors check the box it allows me to add commits directly to the PR (which your GitHub Desktop client will pull down automagically while open). When contributors don't check the box I have to take the extra time to submit a PR to their branch, which they typically don't notice, so I have to remind them, and then I have to check in on the PR periodically to see if it's in yet so that I can move forward, and so on…

With such a large quantity of PRs and issues coming through I just have to minimize the amount of interaction and maximize the amount of production. It's good rewarding work, but I only wish the volume was a little lower so I could get to all the bigger items on the TODO list, like the new EEPROM data schema, the first-time setup wizard, the auto build configuration tool, and so many more….

Thanks for understanding. See you next time!
